src/pages/blog/neon.md

```diff
@@ -0,0 +1,495 @@
+
+```yaml
+tenant:
+  <<: *default
+  url: <%= ENV["CONNECTION_URL"] %>
+  migrations_paths: db/tenant_migrations
+```
+
+We should source `.env.development` into our terminal so we don't have to repeat all the environment variables:
+
+```bash
+source .env.development
+```
+
+You should also copy `config/environments/production.rb` to `config/environments/tenant.rb` to prevent a load warning.
+
+This will allow us to perform migrations. Now we need to move the migration this made (`db/migrate/<date>_create_notes.rb`) to `db/tenant_migrations/<date>_create_notes.rb`, and we can run the bulk migrator:
+```bash
+neon-bulk-migrator \
+    --migrate-command-name="bin/rails db:migrate RAILS_ENV=tenant" \
+    --rollback-command-name="bin/rails db:rollback RAILS_ENV=tenant" \
+    --main-connection-url=$USERS_DATABASE_URL \
+    --neon-api-key=$NEON_API_TOKEN \
+    --neon-project-id=$NEON_PROJECT_ID \
+    --neon-role-name=$NEON_ROLE_NAME \
+    --neon-role-password=$NEON_ROLE_PASSWORD \
+    --neon-database-name=$USER_DATABASE_NAME \
+    --table-branch-column="database_information->>'branch_id'" \
+    --table-connection-column="database_information->>'database_url'" \
+    --table-id-column="id" \
+    --table-name="users" \
+    --tenant-schema-connection-url=$USERS_DB_SCHEMA_URL
+```
+
+## Building the routes
+
+For the next part, we are going to add `before_action :authenticate_user!` to the application controller. This is because most of our functionality wil rely on it.
+```rb
+class ApplicationController < ActionController::Base
+  ...
+
+  # Ensure the user is authenticated
+  before_action :authenticate_user!
+end
+```
+
+Now we will make the `NotesController`:
+
+```
+➜  note-taking git:(main) rails g controller notes
+      create  app/controllers/notes_controller.rb
+      invoke  erb
+      create    app/views/notes
+      invoke  test_unit
+      create    test/controllers/notes_controller_test.rb
+      invoke  helper
+      create    app/helpers/notes_helper.rb
+      invoke    test_unit
+```
+
+From here, we will add a new index by adding this to `config/routes.rb`:
+
+```rb
+# Make the root the main application page.
+root "notes#all"
+```
+
+and just a simple hello world to `app/views/notes/all.html.erb`:
+
+```html
+<h1>Hello World!</h1>
+```
+
+From here, when you go to `/`, we will see a user login screen. Lets sign up and then we will get our first user. If we open `rails c`, we will be able to see this:
+
+```
+➜  note-taking git:(main) rails c
+Loading development environment (Rails 7.2.1)
+note-taking(dev)> User.first!
+  User Load (204.5ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
+=> #<User id: 1, ...
+```
+
+Now we are going to build the controller for this view. Note that in production you likely would want to add pagination to the accessing of notes to improve page load time. We want to access within the `database` block since this will handle the multi-tenant behaviour:
```

Comment:
```suggestion
Now we are going to build the controller for this view. Note that in production you likely would want to add pagination to the accessing of notes to improve page load time. We want to access within the `database` block since this will handle the db-per-tenant behaviour:
```